File: pbgestfinbowcl/src/app/gestione-crediti/services/segnalazione-corte-conti-service.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { ConfigService } from "src/app/core/services/config.service";
import { SegnalazioneCorteContiVO } from "../commons/dto/segnalazione-corte-contiVO";
import { StoricoSegnalazioneCorteContiDTO } from "../commons/dto/storico-segnalazione-corte-conti-dto";

@Injectable()
export class SegnalazioneCorteContiService{
        constructor(
            private configService: ConfigService,
            private http: HttpClient
        ){}


        getSegnalazione(idSegnalazioneCorteConti: number): Observable<SegnalazioneCorteContiVO>{ 
            let url = this.configService.getApiURL() + '/restfacade/segnalazioneCorteConti/getSegnalazione';
            let params = new HttpParams()
           if(idSegnalazioneCorteConti){
               params = params.set("idSegnalazioneCorteConti", idSegnalazioneCorteConti.toString());
           }     
        return this.http.get<SegnalazioneCorteContiVO>(url, { params: params });
        }

        getStorico(idProgetto: number, idModalitaAgevolazione:number): Observable<Array<StoricoSegnalazioneCorteContiDTO>>{
            let url = this.configService.getApiURL()+ "/restfacade/segnalazioneCorteConti/getStorico"; 
            let params= new HttpParams()
            .set("idProgetto", idProgetto.toString())
            .set("idModalitaAgevolazione",  idModalitaAgevolazione.toString()); 
            return this.http.get<Array<StoricoSegnalazioneCorteContiDTO>>(url, {params: params}); 

        }

        getStoricoSegnalazioniCorteConti (idProgetto: number, idModalitaAgevolazione:number): Observable <Array<StoricoSegnalazioneCorteContiDTO>>{
            let url = this.configService.getApiURL() + '/restfacade/segnalazioneCorteConti/getStoricoSegnalazioni';
            let params = new HttpParams()
            .set("idProgetto",idProgetto.toString())
            .set("idModalitaAgevolazione",  idModalitaAgevolazione.toString());
            return this.http.get<Array<StoricoSegnalazioneCorteContiDTO>>(url, { params: params });
        }

        insertSegnalazione(segnalazioneCorteContiVO: SegnalazioneCorteContiVO, idModalitaAgevolazione:number): Observable<boolean>{
            let url = this.configService.getApiURL()+ "/restfacade/segnalazioneCorteConti/insertSegnalazione";
            let params= new HttpParams().set("idModalitaAgevolazione",  idModalitaAgevolazione.toString());  
            return this.http.post<boolean>(url, segnalazioneCorteContiVO, {params: params}); 
        }

        modificaSegnalazione(segnalazioneCorteContiVO: SegnalazioneCorteContiVO, idSegnalazioneCorteConti: number, idModalitaAgevolazione:number): Observable<boolean>{
           let url = this.configService.getApiURL()+ "/restfacade/segnalazioneCorteConti/modifcaSegnalazione";
           let params =  new HttpParams().set("idSegnalazioneCorteConti", idSegnalazioneCorteConti.toString())
           .set("idModalitaAgevolazione", idModalitaAgevolazione.toString()); 
            return this.http.post<boolean>(url, segnalazioneCorteContiVO, {params: params}); 
        }

        /* insertSegnalazione(segnalazioneCorteContiVO: SegnalazioneCorteContiVO,allegati,idModalitaAgevolazione): Observable<boolean>{
            let url = this.configService.getApiURL()+ "/restfacade/segnalazioneCorteConti/insertSegnalazione"; 
            let params = new HttpParams()
            let formData = new FormData();
            for (var i = 0; i < allegati.length; i++) {
            console.log(allegati[i]);
            console.log(allegati[i].name);
            formData.append("file", allegati[i], allegati[i].name);
            }
            formData.append(
            "segnalazioneCorteContiVO",
            JSON.stringify(segnalazioneCorteContiVO)
            );
            formData.append("idModalitaAgevolazione", idModalitaAgevolazione); 
            return this.http.post<any>(url, formData, { params: params }); 
        }

        modificaSegnalazione(segnalazioneCorteContiVO: SegnalazioneCorteContiVO, idSegnalazioneCorteConti: number,allegati,idModalitaAgevolazione): Observable<boolean>{
           let url = this.configService.getApiURL()+ "/restfacade/segnalazioneCorteConti/modifcaSegnalazione";
           let params = new HttpParams()
            let formData = new FormData();
            for (var i = 0; i < allegati.length; i++) {
            console.log(allegati[i]);
            console.log(allegati[i].name);
            formData.append("file", allegati[i], allegati[i].name);
            }
            formData.append(
            "segnalazioneCorteContiVO",
            JSON.stringify(segnalazioneCorteContiVO)
            );
            formData.append("idModalitaAgevolazione", idModalitaAgevolazione); 
            formData.append("idSegnalazioneCorteConti", idSegnalazioneCorteConti);
            return this.http.post<any>(url, formData, { params: params });
        }
 */
}

File: pbgestfinbo/pbgestfinbo-jar/src/main/java/it/csi/pbandi/pbgestfinbo/business/service/SegnalazioneCorteContiService.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbgestfinbo.business.service;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.Consumes;
import javax.ws.rs.DefaultValue;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.QueryParam;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import it.csi.pbandi.pbgestfinbo.integration.vo.SegnalazioneCorteContiVO;

@Path("/segnalazioneCorteConti")
public interface SegnalazioneCorteContiService {
	
	@GET
	@Path("/getSegnalazione")
	@Produces(MediaType.APPLICATION_JSON)
	Response getSegnalazioneCorteConti(@DefaultValue("0") @QueryParam("idSegnalazioneCorteConti") Long idSegnalazioneCorteConti,
	        @Context HttpServletRequest req);
	
	@POST
	@Path("/insertSegnalazione")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	Response insertSegnalazioneCorteConti(SegnalazioneCorteContiVO segnalazioneCorteContiVO,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev); 
	
	@POST
	@Path("/modifcaSegnalazione")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	Response modifcaSegnalazioneCorteConti(SegnalazioneCorteContiVO segnalazioneCorteContiVO, 
			@DefaultValue("0") @QueryParam("idSegnalazioneCorteConti") Long idSegnalazioneCorteConti,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev);
	
	
	// storico con dt fine validita = null
	@GET
	@Path("/getStoricoSegnalazioni")
	@Produces(MediaType.APPLICATION_JSON)
	Response getStoricoSegnalazioni(
			@DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev,
			@Context HttpServletRequest req);
	
	
	@GET
	@Path("/getStorico")
	@Produces(MediaType.APPLICATION_JSON)
	Response getStorico(@DefaultValue("0") @QueryParam("idProgetto") Long idProgetto,
			@DefaultValue("0") @QueryParam("idModalitaAgevolazione") int idModalitaAgev,
			@Context HttpServletRequest req);

}
